GeeksCoreLibrary/Modules/PostalServices/PostNL/Services/PostNLService.cs

```diff
@@ -0,0 +1,400 @@
+        /// </summary>
+        private async Task CleanLogsAsync()
+        {
+            await databaseConnection.ExecuteAsync($"DELETE FROM {PostNlLogTableName} WHERE datetime < DATE_SUB(NOW(), INTERVAL 2 WEEK)");
+        }
+
+        /// <summary>
+        /// Gets the settings for the specified shipping location
+        /// </summary>
+        /// <param name="shippingLocation"></param>
+        /// <returns></returns>
+        public async Task<SettingsModel> GetSettingsAsync(ShippingLocations shippingLocation)
+        {
+            var result = new SettingsModel();
+            switch (shippingLocation)
+            {
+                case ShippingLocations.Netherlands:
+                    result.CustomerCode = await objectService.FindSystemObjectByDomainNameAsync("PostNlNetherlandsCustomerCode");
+                    result.CustomerNumber = await objectService.FindSystemObjectByDomainNameAsync("PostNlNetherlandsCustomerNumber");
+                    result.BarcodeType = "3S";
+                    result.BarcodeSerie = "000000000-999999999";
+                    result.ProductCode = "3085";
+                    break;
+
+                case ShippingLocations.Europe:
+                    result.CustomerCode = await objectService.FindSystemObjectByDomainNameAsync("PostNlEuropeCustomerCode");
+                    result.CustomerNumber = await objectService.FindSystemObjectByDomainNameAsync("PostNlNetherlandsCustomerNumber");
+                    result.BarcodeType = "3S";
+                    result.BarcodeSerie = "0000000-9999999";
+                    result.ProductCode = "4952";
+                    break;
+
+                case ShippingLocations.Global:
+                    result.CustomerCode = await objectService.FindSystemObjectByDomainNameAsync("PostNlGlobalCustomerCode");
+                    result.CustomerNumber = await objectService.FindSystemObjectByDomainNameAsync("PostNlEuropeCustomerNumber");
+                    result.BarcodeType = "CL";
+                    result.BarcodeSerie = "0000-9999";
+                    result.ProductCode = "4945";
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(shippingLocation), shippingLocation, null);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Function for getting the country short name code based on the specified Wiser item
+        /// </summary>
+        /// <param name="orderDetails">The details of the order</param>
+        /// <param name="entityName">The entity name of the property containing the country code</param>
+        /// <returns>The country code</returns>
+        private async Task<string> GetCountryCodeAsync(WiserItemModel orderDetails, string entityName)
+        {
+            var orderCountryCode = orderDetails.GetDetailValue(entityName);
+            string countryCode;
+
+            if (ulong.TryParse(orderCountryCode, out var countryId))
```

Comment:
Hier graag `UInt64.TryParse` van maken. In onze richtlijnen staat ook dat je voor het aanroepen van (static) functies altijd de CLR types moet gebruiken (dat zijn de types van .NET, zoals `Int32`, `Int64`, `String` (met hoofdletter) etc) en bij variabelen en properties de C# types moet gebruiken (dat zijn de types met kleine letters, zoals `int`, `long`, `string` etc).